src/panels/lovelace/common/long-press.ts

```diff
@@ -0,0 +1,98 @@
+import {render, directive} from 'lit-html';
+
+const isTouch =
+  "ontouchstart" in window ||
+  navigator.MaxTouchPoints > 0 ||
+  navigator.msMaxTouchPoints > 0;
+
+const longPressSetup = () => {
+  if(document.body.longPress) return document.body.longPress;
+
+  const longPress = document.createElement("div");
+  document.body.appendChild(longPress);
+  document.body.longPress = longPress;
+  Object.assign(longPress.style, {
+    position: "fixed",
+    width: isTouch ? "100px" : "50px",
+    height: isTouch ? "100px" : "50px",
+    transform: "translate(-50%, -50%)",
+    pointerEvents: "none",
+  });
+  longPress.ripple = document.createElement("mwc-ripple");
+  longPress.appendChild(longPress.ripple);
+  longPress.ripple.primary = true;
+
+  longPress.timer = null;
+  longPress.held = false;
+
+  longPress.start = (x,y) => {
+    Object.assign(longPress.style, {
+      top: `${y}px`,
+      left: `${x}px`,
+      visibility: 'visible',
+    });
+    longPress.ripple.disabled = false;
+    longPress.ripple.active = true;
+    longPress.ripple.unbounded = true;
+  };
+
+  longPress.stop = () => {
+    longPress.ripple.active = false;
+    longPress.ripple.disabled = true;
+    longPress.style.visibility = "hidden";
+  };
+
+  [
+    isTouch ? "touchcancel" : "mouseout",
+    "mouseup",
+    "touchmove",
+    "mousewheel",
+    "wheel",
+    "scroll",
+  ].forEach((ev) => {
+    document.addEventListener(ev, () => {
+      clearTimeout(longPress.timer);
+      longPress.stop();
+    });
+  });
+
+  return longPress;
+};
+
+export const longPressBind = (element, onClick=null, onHold=null) => {
+  const longPress = longPressSetup();
+  if(!longPress) return;
+
+  const firstSetup = (element.longPress == null);
+
+  element.longPress = {
```

Comment:
No need to set it as property on the element, you will always get passed the same `part`, so you can use `part.value` and `part.setValue`

Reply:
The value seems to be reset on each new render, so this doesn't work.

Also, setting it as a element property allows longPressBind to be called directly - so it can be used outside of lit-html templates as well. That's why I export that too.

Reply:
It is not supposed to be re-set on each render, that's how Lit maintains state. Are you sure you're actually dealing with the same element when value is not set?

Reply:
Adding
```
console.log(part.value.test);
part.setValue({test: "hello"});
```
to the directive just fills the log with `undefined`, with more popping up when rerendering (such as by changing window size).

Reply:
But what about the element? This one: `const surfaceNode = part.committer.element as HTMLElement;` is it the same?

You can verify by using a weakMap.

```ts
// outside directive
const interactionNodes = new WeakMap();
```

```ts
// inside directive
console.log(interactionNodes.get(part.committer.element))
interactionNodes.set(part.committer.element, "yooo")
```